include/libssh2_sftp.h

```diff
@@ -303,9 +303,19 @@ LIBSSH2_API int libssh2_sftp_rename_ex(LIBSSH2_SFTP *sftp,
                            LIBSSH2_SFTP_RENAME_ATOMIC | \
                            LIBSSH2_SFTP_RENAME_NATIVE)
 
+LIBSSH2_API int libssh2_sftp_posix_rename_ex(LIBSSH2_SFTP *sftp,
+                                             const char *source_filename,
+                                             unsigned int srouce_filename_len,
+                                             const char *dest_filename,
+                                             unsigned int dest_filename_len);
+#define libssh2_sftp_posix_rename(sftp, sourcefile, destfile) \
+    libssh2_sftp_posix_rename_ex((sftp), (sourcefile), \
+                                 (unsigned int)strlen(sourcefile), \
+                                 (destfile), (unsigned int)strlen(destfile))
+
```

Comment:
Would it make sense to publish the non-`_ex` flavour exclusively?

It seems to me that existing `_ex` variants were added because the initial implementation missed to provide the size arguments. Speaking of a new API, we provide them right away in the non-_ex API. Making a separate `_ex` one superfluous.

Reply:
In theory, I agree with this.

In practice, however, at Panic, all our code calls the non-`_ex` version because we don't need the flexability of setting the string size, and we don't want to call `strlen()` all the time in our code.

I suspect 99% of users don't want to have to call a function where they provide string lengths, and would be happier calling a function that just takes NULL-terminated strings.

Reply:
Okay, we can stick to this to avoid confusion.